doc/code-submission-guidelines.md

```diff
@@ -4,11 +4,170 @@ We've enacted standards for commits and pull requests to effectively manage the
 time. We expect all code contributed to follow these standards. If your code doesn't follow them, we
 will kindly ask you to resubmit it in the correct format.
 
+- [Code Guidelines](#code-guidelines)
 - [Git Commits](#git-commits)
 - [Submitting a pull request](#submitting-a-pull-request)
 - [Merging a pull request](#merging-a-pull-request)
 - [Squashing Commits](#squashing-everything-into-one-commit)
 
+## Code Guidelines
+
+### Default Export at Top
+
+When coding using [JavaScipt modules](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules), the `default export` should be at the top of the file right after any import statements and module level variables. This ensures that when you open the file the main code path is the first thing you read.
```

Comment:
This seems arbitrary to me, as it doesn't change how the module works/reads. It also won't work when using function expressions:

![Block-scoped variable 'myFunction' used before its declaration](https://user-images.githubusercontent.com/571265/176200468-8aa544cb-6687-4cff-ac73-0a2513ff504d.png)

Reply:
I guess we should clarify, it should be this:

```js
default export myfunction() {
  return "Hello World"
}
```

I'm not sure how to distinguish that though (the technical term for it)

Reply:
Why should we require people not use function expressions (and declarations instead)? I don't think it's worthwhile to police what JS features contributors use.

This whole block seems like a matter of personal preference, which we intentionally got away from by adding Prettier. Prior to that, there was _tons_ of back-and-forth about code style/spacing/quotes/etc, none of which helped axe-core be what it is today.

Reply:
The team has adopted this policy, so I'm just codifying it in our repo as per our team discussion.

Reply:
Right below this, you say:

> If you encounter any code that we maintain that does not put the `default export` at the top, you should update the file to do so.

So I'm not sure this has been adopted.

Again, why should we prefer declarations over expressions? Why does this matter?

Reply:
I don't think this is going to prevent anyone who want to contribute from doing so, and it does make the code base easier to work in. I think a consistent code base is more important than they very occasional "Can you put this function at the top of the file" comment.

Reply:
> I don't think this is going to prevent anyone who want to contribute from doing so

It's not going to prevent contributions, but it is extremely tedious and somewhat discouraging to be told _"This is great, but redo it for an arbitrary reason"_.

> it does make the code base easier to work in

Based on what? It requires people to use function declarations rather than expressions, which are **much** more common since "arrow functions" became a thing. If anything, I think this makes it _harder_.

> I think a consistent code base is more important

It's not consistent at all. There are _tons_ of files that use arrow functions and `export` things at the bottom.